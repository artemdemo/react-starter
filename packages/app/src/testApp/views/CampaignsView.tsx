import React, { useState, useEffect, useContext } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGlobe } from '@fortawesome/free-solid-svg-icons';
import { loadCampaigns } from '../model/campaigns/campaignsReq';
import { Campaigns, TCampaign } from '../containers/Campaigns/Campaigns';
import { useTranslation } from '../../hooks/useTranslation';
import { HttpContext } from '../../contexts/http/HttpContext';

export const CampaignsView: React.FC = () => {
  const [isLoading, setIsLoading] = useState<boolean>(true);
  const [campaigns, setCampaigns] = useState<TCampaign[]>([]);
  const { t } = useTranslation();
  const { httpClient } = useContext(HttpContext);

  useEffect(() => {
    loadCampaigns(httpClient)
      .then((campaigns) => setCampaigns(campaigns))
      .finally(() => setIsLoading(false));
  }, []);

  return (
    <>
      <p>
        <FontAwesomeIcon icon={faGlobe} />
        &nbsp; Campaigns View
      </p>
      {isLoading ? <p>{t('loading')}</p> : <Campaigns items={campaigns} />}
    </>
  );
};
