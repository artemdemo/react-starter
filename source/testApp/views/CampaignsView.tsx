import React, { useState, useEffect, useContext } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGlobe } from '@fortawesome/free-solid-svg-icons';
import { loadCampaigns } from '../model/campaigns/campaignsReq';
import { Campaigns, TCampaign } from '../containers/Campaigns/Campaigns';
import { useTranslation } from '../../hooks/useTranslation';
import { HttpContext } from '../../contexts/http/HttpContext';

export const CampaignsView: React.FC = () => {
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const [campaigns, setCampaigns] = useState<TCampaign[]>([]);
  const { t } = useTranslation();
  const { httpClient } = useContext(HttpContext);

  useEffect(() => {
    setIsLoading(true);
    loadCampaigns(httpClient)
      .then((campaigns) => setCampaigns(campaigns))
      .finally(() => setIsLoading(false));
  }, []);

  const renderLoading = () => {
    if (isLoading) {
      return <p>{t('loading')}</p>;
    }
    return null;
  };

  return (
    <>
      <p>
        <FontAwesomeIcon icon={faGlobe} />
        &nbsp; Campaigns View
      </p>
      {renderLoading()}
      <Campaigns items={campaigns} />
    </>
  );
};
