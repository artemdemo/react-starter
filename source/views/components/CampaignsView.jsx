import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {faGlobe} from "@fortawesome/free-solid-svg-icons";
import { loadCampaigns } from "../../model/campaigns/campaignsReq";
import Campaigns from "../../containers/Campaigns/Campaigns";

class CampaignsView extends React.PureComponent {
    constructor(props) {
        super(props);

        this.state = {
            loading: false,
            campaigns: [],
        };
    }

    componentDidMount() {
        this.setState({ loading: true });
        loadCampaigns()
            .then(campaigns => this.setState({ campaigns }))
            .finally(() => this.setState({ loading: false }));
    }

    renderLoading() {
        if (this.state.loading) {
            return (
                <p>
                    Loading...
                </p>
            );
        }
        return null;
    }

    render() {
        return (
            <>
                <p>
                    <FontAwesomeIcon icon={faGlobe} />
                    &nbsp;
                    Campaigns View
                </p>
                {this.renderLoading()}
                <Campaigns
                    items={this.state.campaigns}
                />
            </>
        );
    }
}

export default CampaignsView;
