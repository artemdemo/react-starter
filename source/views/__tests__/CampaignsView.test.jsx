import React from 'react';
import renderer from 'react-test-renderer';
import BluePromise from 'bluebird';
import CampaignsView from '../components/CampaignsView';

jest.mock('../../components/Icon/Icon');
jest.mock('../../containers/Campaigns/Campaigns');
jest.mock('../../model/campaigns/campaignsReq');

describe('<CampaignsView>', () => {
    const campaignsReqMock = require('../../model/campaigns/campaignsReq');

    // I'm mocking implementation with bluebird in order to allow to use `finally()`
    campaignsReqMock.loadCampaigns.mockImplementation(() => BluePromise.resolve([]));

    it('Render with loading', () => {
        const tree = renderer.create(
            <CampaignsView />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
