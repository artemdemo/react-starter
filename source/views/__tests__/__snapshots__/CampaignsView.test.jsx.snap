// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CampaignsView> Render with loading 1`] = `
Array [
  <p>
    <div
      data-mock="Icon"
    >
      {
  "name": "globe"
}
    </div>
    Â  Campaigns View
  </p>,
  <p>
    Loading...
  </p>,
  <div
    data-mock="Campaigns"
  >
    {
  "items": []
}
  </div>,
]
`;
